ntrans_dataprep.py

```diff
@@ -50,13 +48,31 @@ def count_ngram_frequency(n_to_ngrams):
     )
 
 
-def generate_ngrams():
+def format_corpus_sents(sentence):
+    """
+    Reformats spaces in contracted words. Contracted words in
+    <class 'nltk.corpus.reader.bnc.BNCSentence'> are formatted "it 's" and "ca n't"
+
+    Removes all punctuation except apostrophies, and removes the empty strings left from the deleted
+    apostrophies
+    """
+
+    contraction_index = [index for index, word in enumerate(sentence) if "'" in word]
+    for index in reversed(contraction_index):
+        if index != 0:
+            sentence[index - 1] += sentence.pop(index)
```

Comment:
This modifies the list given as an argument, which can be confusing:

```
>>> a = ["foo", "can", "'t", "do", "bar"]
>>> sentence = ["foo", "can", "'t", "do", "bar"]
>>> combined = format_corpus_sents(sentence)
>>> combined
['foo', "can't", 'do', 'bar']
>>> sentence  # this changed too!
['foo', "can't", 'do', 'bar']
```

Consider adding `sentence = sentence.copy()` to the beginning of the function.